tracer/src/Datadog.Trace/Sampling/AgentSamplingRule.cs

```diff
@@ -38,40 +37,46 @@ internal class AgentSamplingRule : ISamplingRule
 
         public float GetSamplingRate(Span span)
         {
-            Log.Debug("Using the default sampling logic");
-            float defaultRate;
+            var service = span.ServiceName;
```

Comment:
span.ServiceName could be null here too right? Do we need a `?? string.Empty` fallback? 

```suggestion
            var service = span.ServiceName ?? string.Empty;
```

Reply:
Good catch. This code existed already, I just moved it around. It looks like nulls are handled correctly bot both `service` and `env, so I don't know why it treats them differently. I'll make them consistent, thanks.

Reply:
Ahh, nullability checks are enabled, and `span.ServiceName` is not nullable, but `TraceContext.Environment` is.